ext/mysqli/mysqli_api.c

```diff
@@ -811,15 +811,60 @@ PHP_FUNCTION(mysqli_stmt_execute)
 {
 	MY_STMT		*stmt;
 	zval		*mysql_stmt;
+	HashTable	*input_params = NULL;
 #ifndef MYSQLI_USE_MYSQLND
 	unsigned int	i;
 #endif
 
-	if (zend_parse_method_parameters(ZEND_NUM_ARGS(), getThis(), "O", &mysql_stmt, mysqli_stmt_class_entry) == FAILURE) {
+	if (zend_parse_method_parameters(ZEND_NUM_ARGS(), getThis(), "O|h!", &mysql_stmt, mysqli_stmt_class_entry, &input_params) == FAILURE) {
```

Comment:
Even though PDO uses a nullable parameter for `$params`, I think we could consider the possibility to make `[]` the default value.

Reply:
What would be the advantage of this? I prefer nullable because you can pass null to the function, but it's just a personal preference and nothing more. If an empty array would be better then it would be good to know why. Maybe you could reply on the mailing list asking which variant people prefer more.

Reply:
I think it's mainly an implementation detail, so I don't think it's worth bringing up there. The advantage is that it's unambigous what the default value means. If null is also allowed then one might wonder if it is the same as an empty array.

Check https://github.com/php/php-src/pull/6400#discussion_r517349804 for a similar example (it's my review comment should the resolved discussion doesn't show up)